File: go-client-sdk/pkg/models/operations/searchcrmevents.go
package operations

import (
	"github.com/speakeasy-sdks/orb-sdks/go-client-sdk/pkg/models/shared"
)

type SearchCrmEventsQueryParams struct {
	AccessToken string   `queryParam:"style=form,explode=true,name=accessToken"`
	AllFields   *bool    `queryParam:"style=form,explode=true,name=allFields"`
	Cursor      *string  `queryParam:"style=form,explode=true,name=cursor"`
	Limit       *float64 `queryParam:"style=form,explode=true,name=limit"`
}

type SearchCrmEventsRequestBodyFilters struct {
	CreatedTime   *shared.DateFilter    `json:"createdTime,omitempty"`
	Description   *shared.StringFilter  `json:"description,omitempty"`
	EndDateTime   *shared.DateFilter    `json:"endDateTime,omitempty"`
	ID            *shared.StringFilter  `json:"id,omitempty"`
	IsAllDayEvent *shared.BooleanFilter `json:"isAllDayEvent,omitempty"`
	Location      *shared.StringFilter  `json:"location,omitempty"`
	ModifiedTime  *shared.DateFilter    `json:"modifiedTime,omitempty"`
	NativeID      *shared.StringFilter  `json:"nativeId,omitempty"`
	StartDateTime *shared.DateFilter    `json:"startDateTime,omitempty"`
	Subject       *shared.StringFilter  `json:"subject,omitempty"`
	Type          *shared.StringFilter  `json:"type,omitempty"`
}

type SearchCrmEventsRequestBody struct {
	Filters *SearchCrmEventsRequestBodyFilters `json:"filters,omitempty"`
}

type SearchCrmEventsResponseBody struct {
	Events         []shared.Event `json:"events,omitempty"`
	NextPageCursor *string        `json:"nextPageCursor,omitempty"`
}

type SearchCrmEventsRequest struct {
	QueryParams SearchCrmEventsQueryParams
	Request     *SearchCrmEventsRequestBody `request:"mediaType=application/json"`
}

type SearchCrmEventsResponse struct {
	ContentType  string
	StatusCode   int64
	ResponseBody *SearchCrmEventsResponseBody
}

File: typescript-client-sdk/src/sdk/models/operations/searchcrmevents.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class SearchCrmEventsQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=accessToken" })
  accessToken: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=allFields" })
  allFields?: boolean;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=cursor" })
  cursor?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=limit" })
  limit?: number;
}


export class SearchCrmEventsRequestBodyFilters extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=createdTime" })
  createdTime?: shared.DateFilter;

  @SpeakeasyMetadata({ data: "json, name=description" })
  description?: shared.StringFilter;

  @SpeakeasyMetadata({ data: "json, name=endDateTime" })
  endDateTime?: shared.DateFilter;

  @SpeakeasyMetadata({ data: "json, name=id" })
  id?: shared.StringFilter;

  @SpeakeasyMetadata({ data: "json, name=isAllDayEvent" })
  isAllDayEvent?: shared.BooleanFilter;

  @SpeakeasyMetadata({ data: "json, name=location" })
  location?: shared.StringFilter;

  @SpeakeasyMetadata({ data: "json, name=modifiedTime" })
  modifiedTime?: shared.DateFilter;

  @SpeakeasyMetadata({ data: "json, name=nativeId" })
  nativeId?: shared.StringFilter;

  @SpeakeasyMetadata({ data: "json, name=startDateTime" })
  startDateTime?: shared.DateFilter;

  @SpeakeasyMetadata({ data: "json, name=subject" })
  subject?: shared.StringFilter;

  @SpeakeasyMetadata({ data: "json, name=type" })
  type?: shared.StringFilter;
}


export class SearchCrmEventsRequestBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=filters" })
  filters?: SearchCrmEventsRequestBodyFilters;
}


export class SearchCrmEventsResponseBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=events", elemType: shared.Event })
  events?: shared.Event[];

  @SpeakeasyMetadata({ data: "json, name=nextPageCursor" })
  nextPageCursor?: string;
}


export class SearchCrmEventsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: SearchCrmEventsQueryParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: SearchCrmEventsRequestBody;
}


export class SearchCrmEventsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  responseBody?: SearchCrmEventsResponseBody;
}
